pkg/util/utils.go

```diff
@@ -244,3 +250,104 @@ func UpdateK8sResourceAnnotation(client client.Client, ctx *context.Context, k8s
 	}
 	return nil
 }
+
+// some resources has complex index, so set it type to string
+func GenerateID(res_id, prefix, suffix string, index string) string {
+	var id strings.Builder
+	if len(prefix) > 0 {
+		id.WriteString(prefix)
+		id.WriteString("_")
+	}
+
+	id.WriteString(res_id)
+	if len(index) > 0 {
+		id.WriteString("_")
+		id.WriteString(index)
+
+	}
+	if len(suffix) > 0 {
+		id.WriteString("_")
+		id.WriteString(suffix)
+	}
+	return id.String()
+}
+
+func GenerateDisplayName(res_name, prefix, suffix, project, cluster string) string {
+	var name strings.Builder
+	if len(prefix) > 0 {
+		name.WriteString(prefix)
+		name.WriteString("-")
+	}
+	if len(project) > 0 {
+		name.WriteString(project)
+		name.WriteString("-")
+	}
+	if len(cluster) > 0 {
+		name.WriteString(cluster)
+		name.WriteString("-")
+	}
+	name.WriteString(res_name)
+	if len(suffix) > 0 {
+		name.WriteString("-")
+		name.WriteString(suffix)
+	}
+	return name.String()
+}
+
+func GenerateTruncName(limit int, res_name, prefix, suffix, project, cluster string) string {
+	name := GenerateDisplayName(res_name, prefix, suffix, project, cluster)
+	return NormalizeName(name)
```

Comment:
NSX resources names have the below limit for display_name length:
1. segment display_name 80 chars
2. group name + domain name 245 chars
3. other NSX resources display_name 255 chars
We use truncated name to limit the display_name
when creating nsx resources.

so, NormalizeName can handle this limit?

Reply:
NormalizeName could only handle the third, the 1,2 could not be handled by that.

Reply:
in ncp, they used the same function like NormalizeName, just limit is different.